src/main/java/org/folio/fqm/repository/ResultSetRepository.java

```diff
@@ -42,35 +46,70 @@ public class ResultSetRepository {
 
   public List<Map<String, Object>> getResultSet(UUID entityTypeId,
```

Comment:
This method is kind of a mess, but it does drastically improve the performance while still allowing for composite result ids. I figured we can just get it in now and clean it up some later.

Reply:
I wouldn't worry about it too much right now, but I bet just breaking out it into a couple of smaller methods would probably go a long way toward cleaning it up